metecho/api/models.py

```diff
@@ -716,8 +716,9 @@ def __str__(self):
         return self.name
 
     def save(self, *args, **kwargs):
+        obj = super().save(*args, **kwargs)
         self.update_status()
-        return super().save(*args, **kwargs)
+        return obj
 
```

Comment:
[Django 4.1 introduces a change](https://docs.djangoproject.com/en/5.0/releases/4.1/#reverse-foreign-key-changes-for-unsaved-model-instances) where using reverse foreign keys before saving an object raises a `ValueError`. Since the `update_status` method interacts with related objects, as seen in methods like `should_update_review`, we have issues with reverse foreign key lookups.

You're correct that calling `super().save()` to ensure the `Epic` instance has an ID before `update_status` is the right fix. But the way it's written now, the status gets updated but not saved in the database.

```python
    def save(self, *args, **kwargs):
        if not self.id:
            super().save(*args, **kwargs)
        self.update_status()
        return super().save()

```

This code makes sure that if the `Epic` is new, we save it before updating the status. By not repeating `*args, **kwargs` in the final save, we avoid duplicating related objects.